### DISCUSSION TITLE
How to handle change auth.user email verification redirect

### DISCUSSION THREAD

**Discussion by bennik88** [supabase/supabase#8930] [Questions]
## Problem:
I'm not sure how to handle the redirect of the email verification. I'm getting this `https://your-dev-environment/#message=Confirmation+link+accepted.+Please+proceed+to+confirm+link+sent+to+the+other+email` URL which redirects me to my app which is working. But what is the best practice to handle this URL. I'm working with Next.js and I thought of probably `const router = useRouter()`. But there is no hint in the documentation on how to work with the`#` in the domain.
Also, there is no documentation about where to set the redirect link. The `supabase.auth.updateUser`function doesn't accept a redirect link so is it best to change it in the Supabase Dashboard under email templates? ` E.g. "{{ .ConfirmationURL }}"`

## Reproduce:
1. Request updating the email
```js
const email = "NEW_EMAIL_ADDRESS"
const { data, error } = await supabase.auth.updateUser({ email })
```

2. Click on the **Change Email** email verification link in both mails you received.

> Confirm Change of Email
> Follow this link to confirm the update of your email from `fromEmail` to `toEmail`:
> 
> [Change Email](verify-link)

3. Your browser now shows the URL like this.
https://your-dev-environment/#message=Confirmation+link+accepted.+Please+proceed+to+confirm+link+sent+to+the+other+email

**Comment by hbussell** [comment]
Hi,

I'm having similar issues using the flutter sdk (1.0.0-dev.3). 

Calling updateUser adds the new email to the user.email_change field. 
Sends the confirmation email. 
When opening up the email in the app I get an onAuthStateChange event of type SignIn.
I assumed clicking the link would move the email_change into email but nothing happens.
I can see the email_change_token is getting set in the database but it is not returned to the app. 
Both using Secure email change on or off has the same result.

What is the expected process to update user email?

**Comment by maheshj01** [comment]
@dshukertjr Is there a way to listen to the email confirmation events?

I am trying to signup as a new user and when I click the link I get redirected to the app, But I would like to redirect the user to a specific part of the app when the email is confirmed. 

Heres what I tried

```dart
 await _supabase.auth.signUp(
          password: user.password,
          email: user.email,
          emailRedirectTo: '<app package name>://login-callback/',
          data: user.schematoJson());
```

```dart
@override
void initState() {
    final _supabase = auth.supabaseClient;
      
     /// This listener is never invoked
    _subscription = _supabase.auth.onAuthStateChange.listen((data) {
      final session = data.session;
      _supabase.auth.getSessionFromUrl(Uri.parse(_supabase.authUrl));

      if (session != null && !haveNavigated) {
        _responseNotifier.value = _responseNotifier.value.copyWith(
          state: RequestState.done,
        );
      }

  Navigate.pushAndPopAll(context, const LoginPage());
}
```

**Reply by dshukertjr** [reply]
@maheshmnj 
Thanks for the question. What you can do is you could redirect the user to something like `<app package name>://login-callback/confirmation/` using the `emailRedirectTo` parameter. 

You can enable Deep links the way that is described here, and this will allow the app to open different screen depending on the path.
https://docs.flutter.dev/development/ui/navigation/deep-linking#enable-deep-linking-on-android

Once that is done, all you need to have is a route at `/confirmation`. You can use packages like [go_router](https://pub.dev/packages/go_router) to create routes at different paths.

I know this hasn't been documented anywhere, and I apologize for it. I am working on creating a guide and a video on how to handle different auth events within Flutter app currently, and hopefully I can release it soon!

**Reply by maheshj01** [reply]
Thank you so much for the quick reply.
